### ISSUE TITLE
Contours plugin is causing phone restart

### ISSUE THREAD

**Issue by krzysztofjanski** [osmandapp/Osmand#2756]
Hello,

Up to date OsmAnd+ and Contours Plugin from google play.
Either contour lines are enable or disabled in map layers, the phone is restarting when map is rendered after OsmAnd+ is started. The problem disappears after uninstalling the plugin.

**Comment by sonora** [comment]
Works for me, cannot reproduce here.

**Comment by krzysztofjanski** [comment]
If I figure out how, I can send some logs. Repeatability of the crash is very stable.

**Comment by sonora** [comment]
It may be worth uninstalling and re-installing OsmAnd, If there is any data you want to keep, please rename the OsmAnd data folder before un-installing.

If it remains  reproducible. please report device model, Android version, and all related version numbers.

**Comment by krzysztofjanski** [comment]
I tried reinstall.
Tried to get the logs, but nothing is reported during phone crash.

Today I spotted that I didn't disable hill shading. With hill shading disabled and contour lines enabled the app does not crash, but as soon as I enable the hill shading, it freezes for few seconds and finally crashes the phone.

I have Sony Xperia M4 Aqua dual sim.
OsmAnd+ 2.3.5
[ro.build.characteristics]: [default]
[ro.build.date.utc]: [1456217138]
[ro.build.date]: [Tue Feb 23 16:45:38 CST 2016]
[ro.build.description]: [E2333-user 5.0 26.1.B.3.109 3057715782 release-keys]
[ro.build.display.id]: [26.1.B.3.109]
[ro.build.fingerprint]: [Sony/E2333/E2333:5.0/26.1.B.3.109/3057715782:user/release-keys]
[ro.build.host]: [cnbjlx20331]
[ro.build.id]: [26.1.B.3.109]
[ro.build.product]: [E2333]
[ro.build.tags]: [release-keys]
[ro.build.type]: [user]
[ro.build.user]: [hudsonslave]
[ro.build.version.all_codenames]: [REL]
[ro.build.version.base_os]: []
[ro.build.version.codename]: [REL]
[ro.build.version.incremental]: [3057715782]
[ro.build.version.release]: [5.0]
[ro.build.version.sdk]: [21]
[ro.build.version.security_patch]: [2015-11-01]
[ro.product-res-path]: [framework/SemcGenericUxpRes/SemcGenericUxpRes.apk]
[ro.product.board]: [msm8916]
[ro.product.brand]: [Sony]
[ro.product.cpu.abi]: [arm64-v8a]
[ro.product.cpu.abilist32]: [armeabi-v7a,armeabi]
[ro.product.cpu.abilist64]: [arm64-v8a]
[ro.product.cpu.abilist]: [arm64-v8a,armeabi-v7a,armeabi]
[ro.product.device]: [E2333]
[ro.product.locale.language]: [en]
[ro.product.locale.region]: [US]
[ro.product.manufacturer]: [Sony]
[ro.product.model]: [E2333]
[ro.product.name]: [E2333]

**Comment by krzysztofjanski** [comment]
Part of the properties were somehow cut from my previous comment. Here they are:
[ro.build.characteristics]: [default]
[ro.build.date.utc]: [1456217138]
[ro.build.date]: [Tue Feb 23 16:45:38 CST 2016]
[ro.build.description]: [E2333-user 5.0 26.1.B.3.109 3057715782 release-keys]
[ro.build.display.id]: [26.1.B.3.109]
[ro.build.fingerprint]: [Sony/E2333/E2333:5.0/26.1.B.3.109/3057715782:user/release-keys]
[ro.build.host]: [cnbjlx20331]
[ro.build.id]: [26.1.B.3.109]
[ro.build.product]: [E2333]
[ro.build.tags]: [release-keys]
[ro.build.type]: [user]
[ro.build.user]: [hudsonslave]
[ro.build.version.all_codenames]: [REL]
[ro.build.version.base_os]: []
[ro.build.version.codename]: [REL]
[ro.build.version.incremental]: [3057715782]
[ro.build.version.release]: [5.0]
[ro.build.version.sdk]: [21]
[ro.build.version.security_patch]: [2015-11-01]
[ro.product-res-path]: [framework/SemcGenericUxpRes/SemcGenericUxpRes.apk]
[ro.product.board]: [msm8916]
[ro.product.brand]: [Sony]
[ro.product.cpu.abi]: [arm64-v8a]
[ro.product.cpu.abilist32]: [armeabi-v7a,armeabi]
[ro.product.cpu.abilist64]: [arm64-v8a]
[ro.product.cpu.abilist]: [arm64-v8a,armeabi-v7a,armeabi]
[ro.product.device]: [E2333]
[ro.product.locale.language]: [en]
[ro.product.locale.region]: [US]
[ro.product.manufacturer]: [Sony]
[ro.product.model]: [E2333]
[ro.product.name]: [E2333]

**Comment by krzysztofjanski** [comment]
This time should work
```
[ro.build.characteristics]: [default]
[ro.build.date.utc]: [1456217138]
[ro.build.date]: [Tue Feb 23 16:45:38 CST 2016]
[ro.build.description]: [E2333-user 5.0 26.1.B.3.109 3057715782 release-keys]
[ro.build.display.id]: [26.1.B.3.109]
[ro.build.fingerprint]: [Sony/E2333/E2333:5.0/26.1.B.3.109/3057715782:user/release-keys]
[ro.build.host]: [cnbjlx20331]
[ro.build.id]: [26.1.B.3.109]
[ro.build.product]: [E2333]
[ro.build.tags]: [release-keys]
[ro.build.type]: [user]
[ro.build.user]: [hudsonslave]
[ro.build.version.all_codenames]: [REL]
[ro.build.version.base_os]: []
[ro.build.version.codename]: [REL]
[ro.build.version.incremental]: [3057715782]
[ro.build.version.release]: [5.0]
[ro.build.version.sdk]: [21]
[ro.build.version.security_patch]: [2015-11-01]
[ro.product-res-path]: [framework/SemcGenericUxpRes/SemcGenericUxpRes.apk]
[ro.product.board]: [msm8916]
[ro.product.brand]: [Sony]
[ro.product.cpu.abi]: [arm64-v8a]
[ro.product.cpu.abilist32]: [armeabi-v7a,armeabi]
[ro.product.cpu.abilist64]: [arm64-v8a]
[ro.product.cpu.abilist]: [arm64-v8a,armeabi-v7a,armeabi]
[ro.product.device]: [E2333]
[ro.product.locale.language]: [en]
[ro.product.locale.region]: [US]
[ro.product.manufacturer]: [Sony]
[ro.product.model]: [E2333]
[ro.product.name]: [E2333]
```

**Comment by sonora** [comment]
My first guess would be you have a corrupt hillshade file, or maybe one of the hillshade files is stored in a corrupt area of your SD card or memory. Maybe try temporarily moving all hillshade files to another folder, and see if this changes things, then move them back (or re-download them) one by one.

**Comment by krzysztofjanski** [comment]
I tried this yesterday.

I'm wondering why android is crashing not just the app. I don' have much programming experience with android, isn't it that only the app should crash or some VM or sandbox?

**Comment by sonora** [comment]
That's why I guessed it may be a hardware issue, possibly related to the SD-card. I cannot reproduce this at all. And the Contour line plugin itself has not been changed in years.

**Comment by krzysztofjanski** [comment]
My mistake somewhere.

Today I moved the net.osmand.plus folder whithin sd card, removed everything related to osmand. Installed again and it works.
I compared md5 of the old 'Hillshade Poland europe.sqlitedb' and a fresh one and it's different. I don't know whether updated or somehow corrupted. I assume no hw SD card issue as md5 was able to read.

So we can consider this issue as solved, I can use the best navi for android (did very well in many trips :) )

Thank you very much for you time and best regards :)
